pkg/processor/k8sprocessor/kube/client_test.go

```diff
@@ -586,16 +602,22 @@ func TestExtractionRules(t *testing.T) {
 	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			if tc.podOwner != nil {
+				pod.OwnerReferences = []meta_v1.OwnerReference{
+					*tc.podOwner,
+				}
+			}
 			c.Rules = tc.rules
+
 			c.handlePodAdd(pod)
 			p, ok := c.GetPod(PodIdentifier(pod.Status.PodIP))
 			require.True(t, ok)
 
 			assert.Equal(t, len(tc.attributes), len(p.Attributes))
 			for k, v := range tc.attributes {
 				got, ok := p.Attributes[k]
-				assert.True(t, ok)
-				assert.Equal(t, v, got)
+				assert.True(t, ok, "Attribute '%s' not found.", k)
+				assert.Equal(t, v, got, "Value of '%s' should be '%s', but was '%s'.", k, v, got)
```

Comment:
These additional log should be included in the message from `testify`, aren't they?

Reply:
Yes, these messages are displayed when the assertions fail. The attribute name is crucial here - without these additional messages, it's not possible to see which attribute the assertion failed for.

Example output:

```console
$ go test
--- FAIL: TestExtractionRules (0.00s)
    --- FAIL: TestExtractionRules/deployment_name_with_owner_lookup (0.00s)
        client_test.go:632: 
                Error Trace:    client_test.go:632
                Error:          Not equal: 
                                expected: "snug-sts"
                                actual  : "dearest-deploy"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -snug-sts
                                +dearest-deploy
                Test:           TestExtractionRules/deployment_name_with_owner_lookup
                Messages:       Value of 'k8s.deployment.name' should be 'snug-sts', but was 'dearest-deploy'.
FAIL
exit status 1
FAIL    github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/kube   0.610s
```